### ISSUE TITLE
How to change MSN loss to PMSN loss? (from paper "The Hidden Uniform Cluster Prior in Self-Supervised Learning") 

### ISSUE THREAD

**Issue by faris-k** [facebookresearch/msn#20]
I couldn't find an official code release for this paper [arxiv.org/abs/2210.07277](https://arxiv.org/abs/2210.07277), in which an extension is proposed to MSN to allow arbitrary feature priors.

It looks like the main difference is a change of a single term in the loss function. Is that correct? How would I implement the changes mentioned in the PMSN paper?

## Previous MSN loss:
![image](https://user-images.githubusercontent.com/68213622/217393278-6ebe5f59-5ff7-4399-8cd2-2dcfc47b2cec.png)

## New loss: Prior Matching for Siamese Networks, PMSN:
![image](https://user-images.githubusercontent.com/68213622/217393619-9aa13a8a-0c08-4564-9ebc-8d70a90081a6.png)

**Comment by faris-k** [comment]
In case anyone stumbles upon this, the [lightly package](https://github.com/lightly-ai/lightly) has a nice implementation of PMSN. I believe they're using a regularization weight of $\lambda = 1$. 

https://github.com/lightly-ai/lightly/blob/ddfed3c4dc03a8d2722df24bfa537d24ac80bde6/examples/pytorch_lightning/pmsn.py

https://github.com/lightly-ai/lightly/blob/ddfed3c4dc03a8d2722df24bfa537d24ac80bde6/lightly/loss/pmsn_loss.py
